### ISSUE TITLE
OS X: Trailing " does not format

### ISSUE THREAD

**Issue by circa1977** [gitterHQ/desktop#141]
On OS X, if I type a double quote (") and continue typing, it converts to an ldquo. 👍 But if I type " at the end of the text and hit Enter, that one does not, unless I trail it with a space.

" as a final character should update to rdquo to match the opening ldquo on submit.

**Comment by MadLittleMods** [comment]
This is a characteristic of OSX and not the app. See https://github.com/gitterHQ/gitter/issues/521

**Comment by circa1977** [comment]
@MadLittleMods I'm happy that the characters are being updated, but are you saying that every other app where this happens does not replace the closing character if the subsequent key pressed is Enter?

I type an opening " and any subsequent key converts that to ldquo. I type a closing " and hit Enter, and it is *not* converted. Any chance Gitter is hijacking the Enter key there?

**Comment by MadLittleMods** [comment]
I am not a lifetime use of OSX so I am not completely familiar with the behavior but here are a couple places to check.

System Preferences -> Keyboard -> Text -> Use smart quotes and dashes

![](https://i.imgur.com/fj3OmWR.png)

To reiterate the point in  gitterHQ/gitter#521, Edit -> Substitutions -> Smart Quotes, etc

![](https://i.imgur.com/7dhuNwB.png)

---

What do you mean by hijacking the enter key? If you press enter in chat mode, we will send the message.

**Comment by circa1977** [comment]
I get that I can disable the OS X feature. I don't want to disable it. I *want* the feature. The feature is not working correctly in Gitter. The issue you cited to is about disabling the feature.

If the last character in the Gitter message is ", OS X isn't converting it when I hit enter to send the message.

Here's a comparison:

In TextEdit, if I type:

> "This is a string"

And pause, OS X converts the last double quote regardless of a subsequent keystroke.

<img width="394" alt="screen shot 2016-05-11 at 3 06 06 pm" src="https://cloud.githubusercontent.com/assets/49657/15193011/eb5d2ba8-1789-11e6-859f-638e2657ba53.png">

In Gitter, if I do the same and pause, there's no conversion. In Gitter, the conversion only happens *on a subsequent keypress*. If the subsequent keypress is Enter, to send the message in chat mode, the conversion never happens. So I end up with this:

<img width="218" alt="screen shot 2016-05-11 at 3 04 16 pm" src="https://cloud.githubusercontent.com/assets/49657/15192951/a6b25da2-1789-11e6-818c-d16e524d3218.png">

See how I have an opening smartquote, and an unconverted endquote? That makes me look goofy.

If I had hit space after the closing character, it'd convert. By pressing Enter to send, it did not.

**Comment by MadLittleMods** [comment]
@circa1977 How does it work for you in other textareas on the web? https://jsfiddle.net/wgwz9f2p/

**Comment by circa1977** [comment]
It doesn't do anything. I just typed in the JS Fiddle fields and in the Gitter web client. They don't bother convert. I'm citing Gitter Desktop, which I compared to TextEdit on the desktop. OS X doesn't modify the characters in Firefox or Safari.

I can't disable the OS feature just for Gitter Desktop...and in Gitter Desktop, unlike other desktop apps, it behaves inconsistently, which affects the presentation of the messages I'm sending people.
